DRAFT task(functional-tests): Add missing reset password tests in Playwright

vpomerleau opened the pull request:
DRAFT - NOT READY FOR REVIEW

## Because

* We need to enable all tests for react reset password flows while continuing to test backbone reset password flows (still in production)

## This pull request

* Add playwright test for backbone reset password route
* Add additional tests for react reset password route
* Add react reset password test for sync

## Issue that this pull request solves

Closes: #FXA-7248

## Checklist

_Put an `x` in the boxes that apply_

- [ ] My commit is GPG signed.
- [ ] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

Any other information that is important to this pull request.

vbudhram commented on packages/functional-tests/tests/react-conversion/resetPassword.spec.ts:
```diff
@@ -56,52 +50,186 @@ test.describe('reset password', () => {
-    expect(
-      await diffPage.locator('text="Your password has been reset"').isEnabled()
-    ).toBeTruthy();
+  test('enter an email with leading whitespace', async ({
```
FWIW, I think `enter an email with leading whitespace`and `enter an email with trailing whitespace` should be unit tests.

vpomerleau replied:
I was wondering about that as I was porting over existing functional tests - feels like there are quite a few grey area where our functional tests could be moved to unit tests. E.g., some spots where we were checking if all page elements were rendered. 

That's part of the reason why I've marked the PR as a draft - I want to circle back on questions like this about what we need to be testing in functional tests for the reset password flow.

vbudhram commented on packages/functional-tests/tests/react-conversion/resetPassword.spec.ts:
```diff
@@ -56,52 +50,186 @@ test.describe('reset password', () => {
+  });
 
-    await page.locator('input[type=email]').fill(credentials.email);
+  test('open /reset_password page, enter unknown email, wait for error', async ({
```
This is another one of those blurry areas, I can see it being a unit test.

vbudhram commented on packages/functional-tests/tests/reset-password/resetPassword.spec.ts:
```diff
@@ -1,12 +1,79 @@
     test.slow();
   });
 
+  test('can reset password', async ({
```
Thank you!

vbudhram commented on packages/functional-tests/pages/resetPassword.ts:
```diff
@@ -28,6 +27,14 @@ export class ResetPasswordPage extends BaseLayout {
   }
 
+  async beginResetPasswordHeadingReact() {
+    const beginResetPasswordHeading = this.page.getByRole('heading', {
```
Not really sure how I feel about this. It seems like it is a completely new page and might be btter in its own class?

vbudhram replied:
New class for the React version of this page*

vpomerleau commented:
Renamed branch FXA-7248-react-reset-pwd-funct-tests - there was an error with the issue number. I'll reopen a PR from the new branch once I've addressed comments.
